Musico/app/controllers/songs_controller.rb

```diff
@@ -0,0 +1,64 @@
+class SongsController < ApplicationController
+  before_action :set_song, only: %i[ show edit update destroy ]
+
+  # GET /songs or /songs.json
+  def index
+      @songs = Song.all
+  end
+
+  # GET /songs/1 or /songs/1.json
+  def show
+  end
+
+  # GET /songs/new
+  def new
+      @song = Song.new
+      @categories = Category.all
+      @languages = Language.all
+      @users = User.all
+
+  end
+
+  # GET /songs/1/edit
+  def edit
+  end
+
+  def create
+      @song = Song.new(song_params)
+      if @song.save
+        @song.user_ids = params[:song][:users].map{ |id| id.to_i }
+        @song.users << current_user
+        redirect_to song_url(@song), notice: "Song was successfully created."
+      else
+        redirect_to new_song_path, notice: "Something went wrong"
+      end 
+  end
+
+  # PATCH/PUT /songs/1 or /songs/1.json
+  def update
+      redirect_to song_url(@song), notice: "Song was successfully updated."
+  end
+
+  # DELETE /songs/1 or /songs/1.json
+  def destroy
+      @song.destroy!
+
+      redirect_to songs_url, notice: "Song was successfully destroyed."
+  end
+
+  def search_user
+      query = params[:query]
+      @users = User.where("name LIKE ? and id <> ?", "%" + query + "%" , current_user.id )
```

Comment:
convert this into a scope in user, where ever possible do not use raw sql. like here for id, you can use `where.not(id: user_id)`

Reply:
done